### DISCUSSION TITLE
Generating complex geometries - Regarding

### DISCUSSION THREAD

**Discussion by dineshkumarreddyg36** [xcompact3d/Incompact3d#99] [Q&A] [answered]
Is it possible to use a mesh file that is generated in some external software  ?

**Comment by mathrack** [comment]
You might want to check [https://github.com/fschuch/xcompact3d_toolbox](https://github.com/fschuch/xcompact3d_toolbox) See [https://xcompact3d-toolbox.readthedocs.io/en/stable/examples/Cylinder.html](https://xcompact3d-toolbox.readthedocs.io/en/stable/examples/Cylinder.html).

**Reply by dineshkumarreddyg36** [reply]
thank you.... @mathrack

**Comment by slaizet** [comment, marked as answer]
As an added comment, the mesh should always be a Cartesian mesh with regular meshes (except in the Y direction where a stretching can be used). As a result, there is no need to import a mesh file, except if you have a complex geometry inside the computational domain (see the links provided by  @mathrack).

Best,

Sylvain
